core/shim/cmd/main.go

```diff
@@ -26,7 +30,17 @@ func run() int {
 	cmd := exec.Command(name, args...)
 	cmd.Env = os.Environ()
 
-	cmd.Stdin = nil
+	if stdinFile, err := os.Open(stdinPath); err == nil {
+		defer stdinFile.Close()
+		cmd.Stdin = stdinFile
+	} else {
+		cmd.Stdin = nil
+	}
+
+	stdoutRedirect, found := internalEnv("_DAGGER_REDIRECT_STDOUT")
+	if found {
+		stdoutPath = stdoutRedirect
```

Comment:
I'm up for a debate on this! I also wasn't sure what to do, so I'll just share my thought process.

I decided not to multiplex since that would mean having a complex `io.Writer` rather than a regular file descriptor, so there'd be a bit of performance overhead. It's also called "redirect" so I think it's fair to say it's no longer visible in the usual location, just like shell redirection.

> Another variation would be to change the meta mount logic to read from the file that is redirected to. But that might be more complicated (also doesn't handle extremely obscure cases like redirecting it to a not normal file like one under `/dev` but I think we can safely ignore that)

Also considered doing this but we would have to keep the file paths somehow. We could put it in the `ContainerID` payload, similar to how we keep `Meta` in there, but again I figured since it's "redirect" it's excusable to do the simpler thing here which is to not make it available.

Reply:
Sounds good, if users complain that they are surprised by this it's easy to change in the future